lib/normalize_grammar.re

```diff
@@ -1,5 +1,62 @@
-open Common;
+module A = Ast_grammar;
+module B = Ast_grammar_normalized;
 
-let normalize = _ast =>
-  raise (Todo);
-  
\ No newline at end of file
+let gensym = () => {
+  /* TODO make it static variable */
+  let static_ident_counter = 1;
+  "intermediate" ++ string_of_int(static_ident_counter)
+}
+
+let rec normalize_to_atom = body => {
+  switch (body) {
+  | A.TOKEN => (B.TOKEN, [])
+  | A.SYMBOL(name)=> (B.SYMBOL(name), [])
+  /* Create intermediate symbol */
+  | _ => {
+    let fresh_ident = gensym();
+    (B.SYMBOL(fresh_ident), [(fresh_ident, fst(normalize_body(body)))]);
```

Comment:
I don't think this will work, you can't ignore snd(normalize_body(body));

In fact I think you should return a Ast_grammar.rule, not an Ast_Grammar_normalized.rule, and in the caller
call recursively the function on it.